komodo-core/src/main/java/org/komodo/repository/RepositoryImpl.java

```diff
@@ -570,6 +587,36 @@ public RepositoryImpl( final Type type,
         this.id = id;
     }
 
+    protected void checkSecurity(UnitOfWork transaction, String nodePath) throws KException {
+        String userWksp = komodoWorkspacePath(transaction);
+
+        if (isSystemTx(transaction))
+            return; // System can do what it wishes
+
+        if (! nodePath.startsWith(WORKSPACE_ROOT))
+            return; // Adding node somewhere other than workspace
```

Comment:
Adding a teiid instance node to the set of teiids, adding to the library. Basically, we don't know enough about what there things look like to tie them down just yet. The API si still very immature for these while the relational API for the workspace is far more mature.